test/integration/cninode/cninode_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cninode_test
+
+import (
+	"time"
+
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/config"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/utils"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/test/framework/resource/k8s/node"
+	testUtils "github.com/aws/amazon-vpc-resource-controller-k8s/test/framework/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[CANARY]CNINode test", func() {
+	Describe("CNINode count verification on adding or removing node", func() {
+		var oldDesiredSize int64
+		var oldMinSize int64
+		var oldMaxSize int64
+		var newSize int64
+		var asgName string
+		BeforeEach(func() {
+			By("getting autoscaling group name")
+			asgName = ListNodesAndGetAutoScalingGroupName()
+			asg, err := frameWork.AutoScalingManager.DescribeAutoScalingGroup(asgName)
+			Expect(err).ToNot(HaveOccurred())
+			oldDesiredSize = *asg[0].DesiredCapacity
+			oldMinSize = *asg[0].MinSize
+			oldMaxSize = *asg[0].MaxSize
+		})
+		AfterEach(func() {
+			By("restoring ASG minSize & maxSize after test")
+			err := frameWork.AutoScalingManager.UpdateAutoScalingGroup(asgName, oldDesiredSize, oldMinSize, oldMaxSize)
+			Expect(err).ToNot(HaveOccurred())
+			// sleep to allow ASG to be updated
+			time.Sleep(testUtils.ResourceCreationTimeout)
+		})
+
+		Context("when new node is added", func() {
+			It("it should create new CNINode", func() {
+				newSize = oldDesiredSize + 1
+				// Update ASG to set desiredSize
+				By("adding new node")
+				err := frameWork.AutoScalingManager.UpdateAutoScalingGroup(asgName, newSize, oldMinSize, newSize)
+				Expect(err).ToNot(HaveOccurred())
+				// sleep to allow ASG to be updated
+				time.Sleep(testUtils.ResourceCreationTimeout)
```

Comment:
Instead of hard coded waiting, we should use polling to avoid test flakiness.

Reply:
Added `WaitTillNodeSizeUpdated` to check that node size is updated to desired capacity of the autoscaling group.